src/app/components/List/EBookList.jsx

```diff
@@ -62,25 +86,37 @@ const arrowRenderer = (open) => {
   return <i className={arrowClass} />;
 };
 
-// gets all links from all ebooks, classified on download true or false
-const linksArray = ({ ebooks, download }) => unique(
-  flattenDeep(ebooks.filter(item => item.links).map(item => item.links)) //
-    .filter(link => link.download === download), //
-  'url',
-);
-// const linksArray = ({ ebooks, download }) => flattenDeep(ebooks.map(item => item.links)).filter(link => link.download === download);
+// gets all links from all ebooks
+// First searches for eReader links
+// If it can't find then, it classifies them on download
+// TODO: This method does not work on items that have both eReader and non-eReader links
```

Comment:
Just want to make sure this is an OK assumption to make, especially for now.

Reply:
I think its an okay assumption to make for the time being, but I think there is a way to implement this so that we can combine the two. Really what we want to filter out are the `content.opf` links, which aren't used right now, right?

The simplest logic to check for that seems to be `link.ereader === true || link.local === false`, which should filter out non-ereader local files, if I'm thinking about this correctly. Then you could just return that array without having the extra `flattenDeep` call and get both types of links together.

Not 100% sure about that and we can obviously put this off for another branch/ticket if we feel that it merits closer attention or a change to the flags that are getting set.

Reply:
Wait that logic would actually be `link.ereader === true || (link.local === false && link.download === download)`, which is probably a sign that this merits more thought